### ISSUE TITLE
Merge Armoria API with Armoria UI

### ISSUE THREAD

**Issue by Azgaar** [Azgaar/armoria-api#6]
There is no real need in Armoria API being a separate project. The maintenance has a cost (both time and money), while a separate project doesn't provide any value.

We can create a single SvelkeKit project and move the API there as well as UI. Can deploy for free on Vercel

Issues:
* it will take time, at least 1-2 weeks
* url will charge, so need to do a fix in FMG and Watabou's City Generator (and other users if any)
* Armoria API is currently hosted on Heroku. We would need to add a redirect there for some time, but not long as Heroku is not free

Benefits:
* Single codebase, easy to manage
* Close to 0 maintenance for the API
* Free hosting
* Same infrastructure and framework as for Deorum

**Comment by Azgaar** [comment]
@Blipz, what do you think about it?

**Comment by Blipz** [comment]
> @Blipz, what do you think about it?

Always nice to reduce code duplication! If I understand correctly, renderers for UI and API would still be different, right?

**Comment by Azgaar** [comment]
Yes, node.js cannot render as browser it doesn't have canvas API. So it will be basically the same

**Comment by Blipz** [comment]
> Yes, node.js cannot render as browser it doesn't have canvas API. So it will be basically the same

I'd suggest replacing resvg-js by a package using a headless browser, like https://www.npmjs.com/package/convert-svg-to-png. A quick test shows that this would solve all the feature inconsistencies between UI and API.

**Comment by Azgaar** [comment]
> > Yes, node.js cannot render as browser it doesn't have canvas API. So it will be basically the same
> 
> I'd suggest replacing resvg-js by a package using a headless browser, like https://www.npmjs.com/package/convert-svg-to-png. A quick test shows that this would solve all the feature inconsistencies between UI and API.

Sounds interesting. Did you try it? I have a concern on how fast it can be, it should be pretty fast as it's an API.

**Comment by Blipz** [comment]
> Sounds interesting. Did you try it? I have a concern on how fast it can be, it should be pretty fast as it's an API.

Here are some metrics on my machine:
- resvg-js runs in about 30ms
- convert-svg-to-png runs in about 200ms if the converter instance is kept, or 1s if it is not

**Comment by Azgaar** [comment]
200ms is acceptable, I believe. Yes, it would be nice to use chromium engine to render SVG, it will allow us to remove the code difference between 2 renderers.

**Comment by Blipz** [comment]
It turns out the performance depends quite heavily on the size of the image (previous metrics were for 500x500), so I don't know how viable it will be in the end.

**Comment by Azgaar** [comment]
500 is a good size. We can set a hard-coded limit to 1000px. If someone needs more, they will have to use SVG and render it manually at any size
